sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -437,61 +437,76 @@ object DataSourceStrategy {
     }
   }
 
+  /**
+   * Find the column name of an expression that can be pushed down.
+   */
+  private[sql] def pushDownColName(e: Expression): Option[String] = {
```

Comment:
`private[sql]` would better be removed [SPARK-16964](https://issues.apache.org/jira/browse/SPARK-16964)

Reply:
Since it's not an instance that will be used somewhere else, and mainly for internal usage, I would like to tighten the visibility.

Reply:
I understand `private[sql]` can scope explicitly which makes sense in a way. However, the current decision was made across the codebase as of SPARK-16964 which also makes sense. What about sticking to one existing way as the current codebase does, and changing it globally later if this is found problematic?

Reply:
+1 for removal.